### PR TITLE
Update NPM package dependencies

### PR DESCRIPTION
**Pull request by abozhinov**
Hello, 
I have dependency errors because using newest postcss, less, stylus loaders and for that we need to allow them in the package.json file. This PR will not break anything and it's pretty safe to merge it.

### CODE REVIEW

**Comment by weaverryan** [comment]
Can you also run `yarn install` and commit the updated `yarn.lock` file? That may be related to the broken tests.

File: package.json [hunk 1]
```diff
@@ -119,13 +119,13 @@
...
-    "less-loader": "^11.0.0",
+    "less-loader": "^11.0.0 || ^12.0.0",
     "postcss": "^8.3.0",
-    "postcss-loader": "^7.0.0",
+    "postcss-loader": "^7.0.0 || ^8.0.0",
     "sass": "^1.17.0",
     "sass-loader": "^13.0.0 || ^14.0.0",
-    "stylus": "^0.58.1",
-    "stylus-loader": "^7.0.0",
+    "stylus": "^0.60.0 || ^0.62.0",
```
Comment (stof) [inline package.json:127]: reducing the range of a peer dependency is a BC break. So you should not remove support for version 0.58

**Review by stof — CHANGES_REQUESTED** [review]

**Comment by abozhinov** [comment]
I've update the package.json and yarn.lock

**Comment by weaverryan** [comment]
Tests still failing - I think we actually need to go into the 3 `yarn-pnp-*` directories here - https://github.com/symfony/webpack-encore/tree/main/test_apps -and run `yarn install` in each and commit that updated lock file. I know, annoying - but we test a lot of different situations.

**Comment by vinceAmstoutz** [comment]
@abozhinov Do y ou need any help?

**Comment by abozhinov** [comment]
I run yarn install on each yarn-pnp-* directory but didn't help.

**Comment by abozhinov** [comment]
I don't know why the yarn.lock files was changed... I've recreate it with the newest packages ...
➤ YN0028: The lockfile would have been modified by this install, which is explicitly forbidden.

**Comment by karpilin** [comment]
@abozhinov this is because of last weeks yarn 4.1 release. I'll make separate PR to update the test_app setups.

Meanwhile, I updated the lockfiles and I've made PR into your fork https://github.com/abozhinov/webpack-encore/pull/1

**Comment by karpilin** [comment]
@stof @weaverryan is there anything that needs correcting before this PR can be merged in?

**Comment by vinceAmstoutz** [comment]
@stof @weaverryan ?

**Comment by vinceAmstoutz** [comment]
@fabpot @nicolas-grekas ?

**Comment by rugbymauri** [comment]
Works for me

```
/var/www $ npm list
www@ /var/www
+-- @babel/core@7.24.7
+-- @babel/preset-env@7.24.7
+-- @hotwired/stimulus@3.2.2
+-- @hotwired/turbo@8.0.4
+-- @symfony/stimulus-bridge@3.2.2
+-- @symfony/ux-autocomplete@npm:assets@1.0.0 -> ./vendor/symfony/ux-autocomplete/assets
+-- @symfony/ux-dropzone@npm:assets@1.1.0 -> ./vendor/symfony/ux-dropzone/assets
+-- @symfony/ux-live-component@npm:assets@1.0.0 -> ./vendor/symfony/ux-live-component/assets
+-- @symfony/ux-turbo@npm:assets@0.1.0 -> ./vendor/symfony/ux-turbo/assets
+-- @symfony/webpack-encore@4.6.1 (git+ssh://git@github.com/abozhinov/webpack-encore.git#8cb9590417074245f90a5db51ca03f7d448cabbf)
+-- @tailwindcss/forms@0.5.7
+-- @zackad/prettier-plugin-twig-melody@0.6.0
+-- autoprefixer@10.4.19
+-- core-js@3.37.1
+-- file-loader@6.2.0
+-- postcss-loader@8.1.1
+-- postcss@8.4.38
+-- prettier@3.3.2
+-- quill@2.0.2
+-- regenerator-runtime@0.14.1
+-- sass-loader@14.2.1
+-- sass@1.77.5
+-- stimulus-textarea-autogrow@4.1.0
+-- tailwindcss-stimulus-components@5.0.2
+-- tailwindcss@3.4.4
+-- tom-select@2.3.1
+-- webpack-cli@5.1.4
+-- webpack-notifier@1.15.0
`-- webpack@5.92.0


```

```
/var/www $ composer show
araise/definition-toolbox                      dev-main ef6068c  Definitions and more
bamarni/composer-bin-plugin                    1.8.2             No conflicts for your bin dependencies
behat/mink                                     1.11.0            Browser controller/emulator abstraction for PHP
brick/math                                     0.12.1            Arbitrary-precision arithmetic library
clue/ndjson-react                              1.3.0             Streaming newline-delimited JSON (NDJSON) parser and encoder for ReactPHP.
composer/pcre                                  3.1.4             PCRE wrapping library that offers type-safe preg_* replacements.
composer/semver                                3.4.0             Semver library that offers utilities, version constraint parsing and validation.
composer/xdebug-handler                        3.0.5             Restarts a process without Xdebug.
dbrekelmans/bdi                                1.3.0             PHAR distribution of dbrekelmans/browser-driver-installer.
dealerdirect/phpcodesniffer-composer-installer 1.0.0             PHP_CodeSniffer Standards Composer Installer Plugin
doctrine/cache                                 2.2.0             PHP Doctrine Cache library is a popular cache implementation that supports many different dr...
doctrine/collections                           2.2.2             PHP Doctrine Collections library that adds additional functionality on top of PHP arrays.
doctrine/data-fixtures                         1.7.0             Data Fixtures for all Doctrine Object Managers
doctrine/dbal                                  4.0.3             Powerful PHP database abstraction layer (DBAL) with many features for database schema intros...
doctrine/deprecations                          1.1.3             A small layer on top of trigger_error(E_USER_DEPRECATED) or PSR-3 logging with options to di...
doctrine/doctrine-bundle                       2.12.0            Symfony DoctrineBundle
doctrine/doctrine-fixtures-bundle              3.6.1             Symfony DoctrineFixturesBundle
doctrine/doctrine-migrations-bundle            3.3.1             Symfony DoctrineMigrationsBundle
doctrine/event-manager                         2.0.1             The Doctrine Event Manager is a simple PHP event system that was built to be used with the v...
doctrine/inflector                             2.0.10            PHP Doctrine Inflector is a small library that can perform string manipulations with regard ...
doctrine/instantiator                          2.0.0             A small, lightweight utility to instantiate objects in PHP without invoking their constructors
doctrine/lexer                                 3.0.1             PHP Doctrine Lexer parser library that can be used in Top-Down, Recursive Descent Parsers.
doctrine/migrations                            3.7.4             PHP Doctrine Migrations project offer additional functionality on top of the database abstra...
doctrine/orm                                   3.2.0             Object-Relational-Mapper for PHP
doctrine/persistence                           3.3.2             The Doctrine Persistence project is a set of shared interfaces and functionality that the di...
doctrine/sql-formatter                         1.4.0             a PHP SQL highlighting library
egulias/email-validator                        4.0.2             A library for validating emails against several RFCs
evenement/evenement                            3.0.2             Événement is a very simple event dispatching library for PHP
fakerphp/faker                                 1.23.1            Faker is a PHP library that generates fake data for you.
fidry/cpu-core-counter                         1.1.0             Tiny utility to get the number of CPU cores.
friendsofphp/php-cs-fixer                      3.58.1            A tool to automatically fix PHP code style
guzzlehttp/guzzle                              7.8.1             Guzzle is a PHP HTTP client library
guzzlehttp/promises                            2.0.2             Guzzle promises library
guzzlehttp/psr7                                2.6.2             PSR-7 message implementation that also provides common utility methods
jms/metadata                                   2.8.0             Class/method/property metadata management in PHP
knpuniversity/oauth2-client-bundle             2.18.1            Integration with league/oauth2-client to provide services
kubawerlos/php-cs-fixer-custom-fixers          3.21.0            A set of custom fixers for PHP CS Fixer
league/flysystem                               3.28.0            File storage abstraction for PHP
league/flysystem-bundle                        3.3.5             Symfony bundle integrating Flysystem into Symfony 5.4+ applications
league/flysystem-local                         3.28.0            Local filesystem adapter for Flysystem.
league/mime-type-detection                     1.15.0            Mime-type detection for Flysystem
league/oauth2-client                           2.7.0             OAuth 2.0 Client Library
league/oauth2-google                           4.0.1             Google OAuth 2.0 Client Provider for The PHP League OAuth2-Client
masterminds/html5                              2.9.0             An HTML5 parser and serializer.
monolog/monolog                                3.6.0             Sends your logs to files, sockets, inboxes, databases and various web services
myclabs/deep-copy                              1.12.0            Create deep copies (clones) of your objects
nikic/php-parser                               5.0.2             A PHP parser written in PHP
paragonie/random_compat                        9.99.100          PHP 5.x polyfill for random_bytes() and random_int() from PHP 7
patchlevel/event-sourcing                      3.0.0             A lightweight but also all-inclusive event sourcing library with a focus on developer experi...
patchlevel/event-sourcing-bundle               3.0.x-dev 8975876 symfony bundle for patchlevel/event-sourcing
patchlevel/hydrator                            1.3.2             Hydrator
patchlevel/worker                              1.2.0             Gives the opportunity to build a stable worker that terminates properly when limits are exce...
phar-io/manifest                               2.0.4             Component for reading phar.io manifest information from a PHP Archive (PHAR)
phar-io/version                                3.2.1             Library for handling version information and constraints
php-webdriver/webdriver                        1.15.1            A PHP client for Selenium WebDriver. Previously facebook/webdriver.
phpdocumentor/reflection-common                2.2.0             Common reflection classes used by phpdocumentor to reflect the code structure
phpdocumentor/reflection-docblock              5.4.1             With this component, a library can provide support for annotations via DocBlocks or otherwis...
phpdocumentor/type-resolver                    1.8.2             A PSR-5 based resolver of Class names, Types and Structural Element Names
phpstan/phpdoc-parser                          1.29.1            PHPDoc parser with support for nullable, intersection and generic types
phpstan/phpstan                                1.11.4            PHPStan - PHP Static Analysis Tool
phpstan/phpstan-doctrine                       1.4.3             Doctrine extensions for PHPStan
phpstan/phpstan-symfony                        1.4.4             Symfony Framework extensions and rules for PHPStan
phpunit/php-code-coverage                      11.0.3            Library that provides collection, processing, and rendering functionality for PHP code cover...
phpunit/php-file-iterator                      5.0.0             FilterIterator implementation that filters files based on a list of suffixes.
phpunit/php-invoker                            5.0.0             Invoke callables with a timeout
phpunit/php-text-template                      4.0.0             Simple template engine.
phpunit/php-timer                              7.0.0             Utility class for timing
phpunit/phpunit                                11.2.1            The PHP Unit Testing framework.
psr/cache                                      3.0.0             Common interface for caching libraries
psr/clock                                      1.0.0             Common interface for reading the clock.
psr/container                                  2.0.2             Common Container Interface (PHP FIG PSR-11)
psr/event-dispatcher                           1.0.0             Standard interfaces for event handling.
psr/http-client                                1.0.3             Common interface for HTTP clients
psr/http-factory                               1.1.0             PSR-17: Common interfaces for PSR-7 HTTP message factories
psr/http-message                               2.0               Common interface for HTTP messages
psr/link                                       2.0.1             Common interfaces for HTTP links
psr/log                                        3.0.0             Common interface for logging libraries
psr/simple-cache                               3.0.0             Common interfaces for simple caching
ralouphie/getallheaders                        3.0.3             A polyfill for getallheaders.
ramsey/collection                              2.0.0             A PHP library for representing and manipulating collections.
ramsey/uuid                                    4.7.6             A PHP library for generating and working with universally unique identifiers (UUIDs).
react/cache                                    1.2.0             Async, Promise-based cache interface for ReactPHP
react/child-process                            0.6.5             Event-driven library for executing child processes with ReactPHP.
react/dns                                      1.13.0            Async DNS resolver for ReactPHP
react/event-loop                               1.5.0             ReactPHP's core reactor event loop that libraries can use for evented I/O.
react/promise                                  3.2.0             A lightweight implementation of CommonJS Promises/A for PHP
react/socket                                   1.15.0            Async, streaming plaintext TCP/IP and secure TLS socket server and client connections for Re...
react/stream                                   1.4.0             Event-driven readable and writable streams for non-blocking I/O in ReactPHP
rector/rector                                  1.1.0             Instant Upgrade and Automated Refactoring of any PHP code
sebastian/cli-parser                           3.0.1             Library for parsing CLI options
sebastian/code-unit                            3.0.0             Collection of value objects that represent the PHP code units
sebastian/code-unit-reverse-lookup             4.0.0             Looks up which function or method a line of code belongs to
sebastian/comparator                           6.0.0             Provides the functionality to compare PHP values for equality
sebastian/complexity                           4.0.0             Library for calculating the complexity of PHP code units
sebastian/diff                                 6.0.1             Diff implementation
sebastian/environment                          7.1.0             Provides functionality to handle HHVM/PHP environments
sebastian/exporter                             6.0.1             Provides the functionality to export PHP variables for visualization
sebastian/global-state                         7.0.1             Snapshotting of global state
sebastian/lines-of-code                        3.0.0             Library for counting the lines of code in PHP source code
sebastian/object-enumerator                    6.0.0             Traverses array structures and object graphs to enumerate all referenced objects
sebastian/object-reflector                     4.0.0             Allows reflection of object attributes, including inherited and non-public ones
sebastian/recursion-context                    6.0.0             Provides functionality to recursively process PHP variables
sebastian/type                                 5.0.0             Collection of value objects that represent the types of the PHP type system
sebastian/version                              5.0.0             Library that helps with managing the version number of Git-hosted PHP projects
slevomat/coding-standard                       8.15.0            Slevomat Coding Standard for PHP_CodeSniffer complements Consistence Coding Standard by prov...
squizlabs/php_codesniffer                      3.10.1            PHP_CodeSniffer tokenizes PHP, JavaScript and CSS files and detects violations of a defined ...
symfony/asset                                  7.1.1             Manages URL generation and versioning of web assets such as CSS stylesheets, JavaScript file...
symfony/browser-kit                            7.1.1             Simulates the behavior of a web browser, allowing you to make requests, click on links and s...
symfony/cache                                  7.1.1             Provides extended PSR-6, PSR-16 (and tags) implementations
symfony/cache-contracts                        3.5.0             Generic abstractions related to caching
symfony/clock                                  7.1.1             Decouples applications from the system clock
symfony/config                                 7.1.1             Helps you find, load, combine, autofill and validate configuration values of any kind
symfony/console                                7.1.1             Eases the creation of beautiful and testable command line interfaces
symfony/css-selector                           7.1.1             Converts CSS selectors to XPath expressions
symfony/debug-bundle                           7.1.1             Provides a tight integration of the Symfony VarDumper component and the ServerLogCommand fro...
symfony/dependency-injection                   7.1.1             Allows you to standardize and centralize the way objects are constructed in your application
symfony/deprecation-contracts                  3.5.0             A generic function and convention to trigger deprecation notices
symfony/doctrine-bridge                        7.1.1             Provides integration for Doctrine with various Symfony components
symfony/doctrine-messenger                     7.1.1             Symfony Doctrine Messenger Bridge
symfony/dom-crawler                            7.1.1             Eases DOM navigation for HTML and XML documents
symfony/dotenv                                 7.1.1             Registers environment variables from a .env file
symfony/error-handler                          7.1.1             Provides tools to manage errors and ease debugging PHP code
symfony/event-dispatcher                       7.1.1             Provides tools that allow your application components to communicate with each other by disp...
symfony/event-dispatcher-contracts             3.5.0             Generic abstractions related to dispatching event
symfony/expression-language                    7.1.1             Provides an engine that can compile and evaluate expressions
symfony/filesystem                             7.1.1             Provides basic utilities for the filesystem
symfony/finder                                 7.1.1             Finds files and directories via an intuitive fluent interface
symfony/flex                                   2.4.5             Composer plugin for Symfony
symfony/form                                   7.1.1             Allows to easily create, process and reuse HTML forms
symfony/framework-bundle                       7.1.1             Provides a tight integration between Symfony components and the Symfony full-stack framework
symfony/http-client                            7.1.1             Provides powerful methods to fetch HTTP resources synchronously or asynchronously
symfony/http-client-contracts                  3.5.0             Generic abstractions related to HTTP clients
symfony/http-foundation                        7.1.1             Defines an object-oriented layer for the HTTP specification
symfony/http-kernel                            7.1.1             Provides a structured process for converting a Request into a Response
symfony/intl                                   7.1.1             Provides access to the localization data of the ICU library
symfony/mailer                                 7.1.1             Helps sending emails
symfony/maker-bundle                           1.60.0            Symfony Maker helps you create empty commands, controllers, form classes, tests and more so ...
symfony/messenger                              7.1.1             Helps applications send and receive messages to/from other applications or via message queues
symfony/mime                                   7.1.1             Allows manipulating MIME messages
symfony/monolog-bridge                         7.1.1             Provides integration for Monolog with various Symfony components
symfony/monolog-bundle                         3.10.0            Symfony MonologBundle
symfony/notifier                               7.1.1             Sends notifications via one or more channels (email, SMS, ...)
symfony/options-resolver                       7.1.1             Provides an improved replacement for the array_replace PHP function
symfony/panther                                2.1.1             A browser testing and web scraping library for PHP and Symfony.
symfony/password-hasher                        7.1.1             Provides password hashing utilities
symfony/phpunit-bridge                         7.1.1             Provides utilities for PHPUnit, especially user deprecation notices management
symfony/polyfill-intl-grapheme                 1.29.0            Symfony polyfill for intl's grapheme_* functions
symfony/polyfill-intl-icu                      1.29.0            Symfony polyfill for intl's ICU-related data and classes
symfony/polyfill-intl-idn                      1.29.0            Symfony polyfill for intl's idn_to_ascii and idn_to_utf8 functions
symfony/polyfill-intl-normalizer               1.29.0            Symfony polyfill for intl's Normalizer class and related functions
symfony/polyfill-mbstring                      1.29.0            Symfony polyfill for the Mbstring extension
symfony/polyfill-php83                         1.29.0            Symfony polyfill backporting some PHP 8.3+ features to lower PHP versions
symfony/process                                7.1.1             Executes commands in sub-processes
symfony/property-access                        7.1.1             Provides functions to read and write from/to an object or array using a simple string notation
symfony/property-info                          7.1.1             Extracts information about PHP class' properties using metadata of popular sources
symfony/routing                                7.1.1             Maps an HTTP request to a set of configuration variables
symfony/runtime                                7.1.1             Enables decoupling PHP applications from global state
symfony/security-bundle                        7.1.1             Provides a tight integration of the Security component into the Symfony full-stack framework
symfony/security-core                          7.1.1             Symfony Security Component - Core Library
symfony/security-csrf                          7.1.1             Symfony Security Component - CSRF Library
symfony/security-http                          7.1.1             Symfony Security Component - HTTP Integration
symfony/serializer                             7.1.1             Handles serializing and deserializing data structures, including object graphs, into array s...
symfony/service-contracts                      3.5.0             Generic abstractions related to writing services
symfony/stimulus-bundle                        2.18.1            Integration with your Symfony app & Stimulus!
symfony/stopwatch                              7.1.1             Provides a way to profile code
symfony/string                                 7.1.1             Provides an object-oriented API to strings and deals with bytes, UTF-8 code points and graph...
symfony/translation                            7.1.1             Provides tools to internationalize your application
symfony/translation-contracts                  3.5.0             Generic abstractions related to translation
symfony/twig-bridge                            7.1.1             Provides integration for Twig with various Symfony components
symfony/twig-bundle                            7.1.1             Provides a tight integration of Twig into the Symfony full-stack framework
symfony/type-info                              7.1.1             Extracts PHP types information.
symfony/ux-autocomplete                        2.18.0            JavaScript Autocomplete functionality for Symfony
symfony/ux-dropzone                            2.18.0            File input dropzones for Symfony Forms
symfony/ux-live-component                      2.18.1            Live components for Symfony
symfony/ux-turbo                               2.18.0            Hotwire Turbo integration for Symfony
symfony/ux-twig-component                      2.18.1            Twig components for Symfony
symfony/validator                              7.1.1             Provides tools to validate values
symfony/var-dumper                             7.1.1             Provides mechanisms for walking through any arbitrary PHP variable
symfony/var-exporter                           7.1.1             Allows exporting any serializable PHP data structure to plain PHP code
symfony/web-link                               7.1.1             Manages links between resources
symfony/web-profiler-bundle                    7.1.1             Provides a development tool that gives detailed information about the execution of any request
symfony/webpack-encore-bundle                  2.1.1             Integration with your Symfony app & Webpack Encore!
symfony/yaml                                   7.1.1             Loads and dumps YAML files
symplify/easy-coding-standard                  12.2.1            Use Coding Standard with 0-knowledge of PHP-CS-Fixer and PHP_CodeSniffer
theseer/tokenizer                              1.2.3             A small library for converting tokenized PHP source code into XML and potentially other formats
twbs/bootstrap-icons                           1.11.3            Official open source SVG icon library for Bootstrap
twig/extra-bundle                              3.10.0            A Symfony bundle for extra Twig extensions
twig/twig                                      3.10.3            Twig, the flexible, fast, and secure template language for PHP
vich/uploader-bundle                           2.3.3             Ease file uploads attached to entities
webmozart/assert                               1.11.0            Assertions to validate method input/output with nice error messages.
whatwedo/php-coding-standard                   1.2.5             Default PHP coding standard at whatwedo
whatwedo/twig-bootstrap-icons                  1.1.2            
zenstruck/assert                               1.5.0             Standalone, lightweight, framework agnostic, test assertion library.
zenstruck/browser                              1.8.1             A fluent interface for your Symfony functional tests.
zenstruck/callback                             1.5.0             Callable wrapper to validate and inject arguments.
zenstruck/foundry                              2.0.1             A model factory library for creating expressive, auto-completable, on-demand dev/test fixtur...
```

**Comment by garak** [comment]
What about eslint? See #1267

**Comment by rugbymauri** [comment]
@garak #1267 is still an issue

**Comment by melroy89** [comment]
Oopsy daisy

**Comment by Kocal** [comment]
Hi everyone, and sorry for the big delay!

Just to let you know, very soon a new version will be tagged with a lot of dependencies updates (see https://github.com/symfony/webpack-encore/milestone/3)

**Pull request closed without merging**